app/src/androidTest/java/com/google/android/perftesting/testrules/MeasureBatteryStats.java

```diff
@@ -135,4 +139,20 @@ public void end(){
             }
         }
     }
-}
\ No newline at end of file
+
+    private void creatPackageNameFile() throws IOException, InterruptedException {
+        if (android.os.Build.VERSION.SDK_INT >= 23) {
```

Comment:
Why is the API level 23 used here? Actually, I don't think the check is necessary.